solidity/contracts/utility/PriceOracle.sol

```diff
@@ -47,15 +39,29 @@ contract PriceOracle is IPriceOracle, Utils {
         tokensToOracles[_tokenB] = _tokenBOracle;
     }
 
-    // ensures that the token is supported by the oracle
-    modifier supportedToken(IERC20Token _token) {
-        _supportedToken(_token);
+    // ensures that the provided addresses are valid
+    modifier vaidAddresses(address _address1, address _address2) {
+        _vaidAddresses(_address1, _address2);
+        _;
+    }
+
+    // error message binary size optimization
+    function _vaidAddresses(address _address1, address _address2) internal pure {
+        _validAddress(_address1);
+        _validAddress(_address2);
+        require(_address1 != _address2, "ERR_SAME_ADDRESSES");
+    }
+
+    // ensures that the provides tokens are supported by the oracle
+    modifier supportedTokens(IERC20Token _tokenA, IERC20Token _tokenB) {
+        _supportedTokens(_tokenA, _tokenB);
         _;
     }
 
     // error message binary size optimization
-    function _supportedToken(IERC20Token _token) internal view {
-        require(tokensToOracles[_token] != address(0), "ERR_UNSUPPORTED_TOKEN");
+    function _supportedTokens(IERC20Token _tokenA, IERC20Token _tokenB) internal view {
+        require(_tokenA != _tokenB, "ERR_SAME_ADDRESSES");

```

Comment:
Why not use `_vaidAddresses(_tokenA, _tokenB)` instead of `require(_tokenA != _tokenB, ...)`?

Reply:
Mostly to avoid the redundant check for `address(0)`. Do you think it'd more readable this way, regarding the small additional gas cost?

Reply:
Depending how frequently it is used. Gas cost precedes all other considerations, that I agree.
But aren't we supposed to check that anyway (that these tokens are not zero)?

Reply:
It is explicitly checked in `tokensToOracles[_tokenA] != address(0)` below

Reply:
we can use the `validAddress` modifier we already have and call it twice.
we can also add a generic modifier that ensures that 2 addresses are different.
thoughts?